File: Code/app/src/main/java/com/group_7/library_management/data/local/entity/NotificationEntity.kt
package com.group_7.library_management.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "notifications")
data class NotificationEntity(
    @PrimaryKey(autoGenerate = true)
    val id: Long?=null,
    val title: String,
    val message: String,
    val time: String,
    val date: String,
    val type: String,
    val isRead: Boolean = false,
    val isDeleted: Boolean = false,
    val createdAt: Long = System.currentTimeMillis()
)

File: backend/src/notifications/entities/notification.entity.ts
export class Notification {
  id: number;
  userId: number;
  title: string;
  message: string;
  type: 'BORROW' | 'RETURN' | 'OVERDUE' | 'SYSTEM' | 'SUPPORT';
  isRead: boolean;
  createdAt: string;
}
